sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -499,14 +499,16 @@ object SQLConf {
       .intConf
       .createWithDefault(40)
 
-  val VECTORIZED_AGG_MAP_MAX_COLUMNS =
-    SQLConfigBuilder("spark.sql.codegen.aggregate.map.columns.max")
+  val ENFORCE_FAST_AGG_MAP_IMPL =
+    SQLConfigBuilder("spark.sql.codegen.aggregate.map.enforce.impl")
       .internal()
-      .doc("Sets the maximum width of schema (aggregate keys + values) for which aggregate with" +
-        "keys uses an in-memory columnar map to speed up execution. Setting this to 0 effectively" +
-        "disables the columnar map")
-      .intConf
-      .createWithDefault(3)
+      .doc("Sets the implementation for fast hash map during aggregation. Could be one of the " +
+        "following: rowbased, vectorized, skip, auto. Defaults to auto, and should only be other " +
+        "values for testing purposes.")
+      .stringConf
+      .transform(_.toLowerCase())
+      .checkValues(Set("rowbased", "vectorized", "skip", "auto"))
+      .createWithDefault("auto")
```

Comment:
Note that this removes the entire vectorized hashmap code from the the test path even though we claim to support it. Let's make sure that we have explicit tests that test for both "rowbased" and "vectorized".